### ISSUE TITLE
Features: Multi-Rename and Find Files

### ISSUE THREAD

**Issue by ags1234** [doublecmd/doublecmd#112]
Thank you for developing such an interesting, useful and free file manager.
There are two not so commonly found options to consider that could improve DC's file managing capability.

1. Make it possible to use a custom timestamp format in Multi-Rename.
When keeping many older versions of the same file in an archive, it is useful to have them renamed after the last modification date, starting with year, then month, then day, followed by a separation character, then time, then separation character, then [N].
Sample name from a document saved today at 16:39: 20210615-163936_report.doc
Free Commander has this option.

2. Make an option to search for long paths and file/folder names. Let the user specify the length in number of characters. Maybe this could be implemented in the Find Files Tool, and in case the user searches for long names, make a button to be able to feed the result list to Multi-Rename Tool for renaming long, troublesome names to shorter ones.
FileBoss searches for long paths and file/folder names: https://www.theutilityfactory.com/summaries/find-long-filenames.htm

**Comment by alexx2000** [comment]
1. No problem, in version 1.0 you can use any date format, see [available placeholders](https://skif-off.github.io/doublecmd.github.io/doc/en/configuration.html#ConfigView).

**Comment by ags1234** [comment]
Using build 9822 now. From Help - About:
-----
Double Commander
Version: 1.0.0 alpha
Revision: 9822
Commit: c5a2222db
Build date: 2021/06/09
Lazarus: 2.0.12.0-64678
FPC: 3.0.4
Platform: x86_64-Win64-qt5
OS version: Windows 10 x86_64

Widgetset library: Qt 5.13.0, libQt5Pas 5.6.1
-----

1. It is exactly what I needed, thank you!

2. This may be a little too advanced for me. Cannot make the script work. Absolutely nothing happens when I press the toolbar button I just made. What I did:
a. read some of the help page
b. making sure I have lua5.1.dll in the same folder as the executable of Double Commander. It was there already.
c. made a toolbar button:
-  cm_ExecuteScript as command
- full path to the script checkfilenamewdx.lua as parameter
![Untitled](https://user-images.githubusercontent.com/21011513/122251654-5f773000-ced3-11eb-9a5b-8e9f124d60ea.png)
